From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Mon, 9 Mar 2015 12:27:25 +0100
Subject: [PATCH] HACK: missing.h: add NDA_*

This is necessary to build with older kernel headers. NDA_VLAN was
introduced in v3.9 and NDA_PORT, NDA_VNI and NDA_IFINDEX in v3.10

This is a hacked version of an upstream commit that doesn't touch
configure.ac.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/shared/missing.h | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/shared/missing.h b/src/shared/missing.h
index 3f93d3226ee3..1ba53e810476 100644
--- a/src/shared/missing.h
+++ b/src/shared/missing.h
@@ -35,6 +35,7 @@
 #include <linux/loop.h>
 #include <linux/audit.h>
 #include <linux/capability.h>
+#include <linux/neighbour.h>
 
 #ifdef HAVE_AUDIT
 #include <libaudit.h>
@@ -623,6 +624,21 @@ static inline int setns(int fd, int nstype) {
 #define CLOCK_BOOTTIME_ALARM 9
 #endif
 
+#if 1
+#define NDA_UNSPEC 0
+#define NDA_DST 1
+#define NDA_LLADDR 2
+#define NDA_CACHEINFO 3
+#define NDA_PROBES 4
+#define NDA_VLAN 5
+#define NDA_PORT 6
+#define NDA_VNI 7
+#define NDA_IFINDEX 8
+#define __NDA_MAX 9
+
+#define NDA_MAX (__NDA_MAX - 1)
+#endif
+
 #ifndef IPV6_UNICAST_IF
 #define IPV6_UNICAST_IF 76
 #endif
